Classes/Controller/FeuserController.php

```diff
@@ -408,7 +408,7 @@ protected function isUserInUserGroups(FrontendUser $user, array $userGroups): bo
     protected function getFollowingUserGroups(int $currentUserGroup): array
     {
         $userGroups = $this->getUserGroupIds();
-        $currentIndex = array_search((int) $currentUserGroup, $userGroups);
+        $currentIndex = array_search((int)$currentUserGroup, array_values($userGroups));
```

Comment:
From my understanding the array_values needs to be used in 410. Else it could happen that the currendIndex is related to a different array index then available in later on.

Reply:
As I understand `array_slice()` does not work respect the array's keys. It only cuts of a slice starting at the given position/offset.

Calling `array_values()` in this line only, will preserve the keys in the returned part of the array (https://www.tehplayground.com/o4ACYYiwt4ach3Ks)